internal/sci/sci.proto

```diff
@@ -1,13 +1,24 @@
 syntax = "proto3";
 
 package sci.v1;
+import "google/protobuf/struct.proto";
 option go_package = "github.com/substratusai/substratus/internal/sci";
 
 service Controller {
   rpc CreateSignedURL(CreateSignedURLRequest) returns (CreateSignedURLResponse) {}
   rpc GetObjectMd5(GetObjectMd5Request) returns (GetObjectMd5Response) {}
+  rpc BindKSAToIAMPrincipal(BindKSAToIAMPrincipalRequest) returns (BindKSAToIAMPrincipalResponse) {} // strictly adds a permission for an IAM principal to a KSA
```

Comment:
Tagging @nstogner 

Our casing is a little inconsistent on other services (`Md5` should be `MD5` if `URL` isn't `Url`) so let's take a moment to tidy those up. Some alternatives to the service name here:

* `AddKSAToIAMPrincipalBinding`
* `CreateKSAToIAMPrincipalBinding`
* `PatchIamPrincipalBinding`
* `AssociateKSAWithPrincipal`

Between Sam and I, we're committed to making this service additive-only and not able to specify an operation (e.g. add, remove).

The Request and Response messages fall out of the naming decision here.

Reply:
`BindIdentity` is short and simple but maybe too much so. What's really happening is establishing a one-way trust relationship between identities. I'm happy to go with `BindIdentity` because this isn't so important - we own the only client of SCI and can change it any time.